src/data_snack/snack.py

```diff
@@ -75,14 +70,12 @@ def set(self, entity: Entity, expire: int = 0) -> Optional[Text]:
         :param expire: number of seconds until the item is expired, or zero for no expiry
         :return: on success returns key used for the object, None on fail
         """
-        cls = entity.__class__
-        type_name = cls.__name__
-        key = self._build_record_key(type_name, entity)
-        record = self._get_serializer(type_name).serialize(entity)
-        if self.connection.set(cls, key, record, expire):
-            return key
+        key = self._build_record_key(entity)
+        record = self._get_serializer(entity.__class__).serialize(entity)
+        if self.connection.set(key, record, expire):
+            return key.keystring  # Should we return keystring or maybe just a key?
 
-    def get(self, cls: Type[Entity], key_values: List[Any]) -> Optional[Entity]:
+    def get(self, cls: Type[Entity], key_values: List[str]) -> Optional[Entity]:
```

Comment:
Key value could be and `int` or `bool` too, it would be better to cast it to str later, I think

Reply:
I don't know why I changed the typing - `Key.keystring` has this string casting. I added a few test cases for Keys.